src/coreclr/ildasm/dasm.cpp

```diff
@@ -7447,6 +7457,41 @@ BOOL DumpFile()
     g_tkEntryPoint = VAL32(IMAGE_COR20_HEADER_FIELD(*g_CORHeader, EntryPointToken)); // integration with MetaInfo
 
 
+    if (g_fR2RNativeManifestMetadata)
+    {
+        //if this is a r2r image, use the native metadata.
+        if( !(g_CORHeader->ManagedNativeHeader.Size >= sizeof(READYTORUN_HEADER) ) )
+        {
+            printError( g_pFile, "/r2rnativemetadata only works on non-composite R2R images." );
```

Comment:
Am I right to assume this is just a shortcut in the implementation (not having to deal with the export symbol used to locate the RTR header in composite images)? Otherwise, composite images certainly do include manifest metadata.

Reply:
R2R composite image manifest metadata doesn't need additional switches to read. I believe it can be read with normal ildasm.  ... Note to self... I should check this.

Reply:
Yep, regular ildasm on a composite image reads the manifest metadata

Reply:
I see, thanks for pointing that out!